app/lib/actions/katello/product/content_create.rb

```diff
@@ -15,10 +16,22 @@ def plan(repository)
                                            label:       repository.custom_content_label,
                                            content_url: content_url(repository))
               content_id = content_create.output[:response][:id]
+
+              katello_content = plan_action(Katello::Content::Create,
+                          name: repository.name,
+                          cp_content_id: content_id,
+                          content_type: repository.content_type,
+                          label: repository.custom_content_label,
+                          content_url: content_url(repository))
+
               plan_action(Candlepin::Product::ContentAdd, owner: repository.product.organization.label,
                                     product_id: repository.product.cp_id,
                                     content_id: content_id)
 
+              plan_action(Katello::ProductContent::Create,
+                          product_id: repository.product.id,
+                          content_id: katello_content.output[:content_id])
+
```

Comment:
Same sort of comment here, we typically just create teh AR objects in th plan phase.  You may consider structuring these such that you have a 👍 
Katello::Content::Create  action that creates the AR record in the plan phase, and then plans Candlepin::Product::ContentCreate which creates it in candlepin in the run phase.